modules/calib3d/src/rhorefc.cpp

```diff
@@ -0,0 +1,2165 @@
+ are permitted provided that the following conditions are met:
+
+   * Redistribution's of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+
+   * Redistribution's in binary form must reproduce the above copyright notice,
+     this list of conditions and the following disclaimer in the documentation
+     and/or other materials provided with the distribution.
+
+   * The name of the copyright holders may not be used to endorse or promote products
+     derived from this software without specific prior written permission.
+
+ This software is provided by the copyright holders and contributors "as is" and
+ any express or implied warranties, including, but not limited to, the implied
+ warranties of merchantability and fitness for a particular purpose are disclaimed.
+ In no event shall the Intel Corporation or contributors be liable for any direct,
+ indirect, incidental, special, exemplary, or consequential damages
+ (including, but not limited to, procurement of substitute goods or services;
+ loss of use, data, or profits; or business interruption) however caused
+ and on any theory of liability, whether in contract, strict liability,
+ or tort (including negligence or otherwise) arising in any way out of
+ the use of this software, even if advised of the possibility of such damage.
+*/
+
+/**
+ * Bilaniuk, Olexa, Hamid Bazargani, and Robert Laganiere. "Fast Target
+ * Recognition on Mobile Devices: Revisiting Gaussian Elimination for the
+ * Estimation of Planar Homographies." In Computer Vision and Pattern
+ * Recognition Workshops (CVPRW), 2014 IEEE Conference on, pp. 119-125.
+ * IEEE, 2014.
+ */
+
+/* Includes */
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include <stddef.h>
+#include <limits.h>
+#include <float.h>
+#include <math.h>
+#include "rhorefc.h"
+
+
+
+/* Defines */
+#define MEM_ALIGN               32
+#define HSIZE                   (3*3*sizeof(float))
+#define MIN_DELTA_CHNG          0.1
+#define REL_CHNG(a, b)          (fabs((a) - (b))/(a))
+#define CHNG_SIGNIFICANT(a, b)  (REL_CHNG(a, b) > MIN_DELTA_CHNG)
+#define CHI_STAT                2.706
+#define CHI_SQ                  1.645
+#define RLO                     0.25
+#define RHI                     0.75
+#define MAXLEVMARQITERS         100
+#define m                       4       /* 4 points required per model */
+#define SPRT_T_M                25      /* Guessing 25 match evlauations / 1 model generation */
+#define SPRT_M_S                1       /* 1 model per sample */
+#define SPRT_EPSILON            0.1     /* No explanation */
+#define SPRT_DELTA              0.01    /* No explanation */
+#define LM_GAIN_LO              0.25    /* See sacLMGain(). */
+#define LM_GAIN_HI              0.75    /* See sacLMGain(). */
+
+
+
+/* For the sake of cv:: namespace ONLY: */
+#ifdef __cplusplus
```

Comment:
#ifndef __cplusplus should be omitted, all the OpenCV code is built with C++ compiler